state/postgresql/v2/postgresql.go

```diff
@@ -89,19 +88,23 @@ func (p *PostgreSQL) Init(ctx context.Context, meta state.Metadata) error {
 		useAzure bool
 		err      error
 	)
-	if meta.Properties["UseAWSIAM"] == "" {
+
+	awsIam, ok := metadata.GetMetadataProperty(meta.Properties, "UseAWSIAM")
```

Comment:
`utils.IsTruthy` please :)

Replace the entire block with this (which handles the case of the property missing already, no need to check for `ok`):

```go
awsIam, _ := metadata.GetMetadataProperty(meta.Properties, "UseAWSIAM")
useAWS = utils.IsTruthy(awsIam)
```